apps/site/pages/en/learn/typescript/introduction.md

```diff
@@ -44,6 +44,49 @@ The first part (with the `type` keyword) is responsible for declaring our custom
 
 There are additional things about this example that you should know. Firstly, if we do not comply with the declared types, TypeScript will inform us that something is wrong and prevent misuse. Secondly, not everything must be typed explicitly—TypeScript infers types for us. For example, the variable `isJustineAnAdult` is of type `boolean` even if we didn't type it explicitly, and `justine` would be a valid argument for our function even though we didn't declare this variable as of `User` type.
 
+## What does TypeScript consist of?
+
+TypeScript consists of two main components: the code itself and type definitions.
+
+### TypeScript Code
+
+The code part is regular JavaScript with additional TypeScript-specific syntax for type annotations. When TypeScript code is compiled, all the TypeScript-specific parts are removed, resulting in clean JavaScript that can run in any environment. For example:
+
+```ts
+// This is TypeScript code
+function greet(name: string) {
+  console.log(`Hello, ${name}!`);
+}
+```
+
+### Type Definitions
+
+Type definitions are declarations that describe the shape of existing JavaScript code. They are usually stored in `.d.ts` files and don't contain any actual implementation—they only describe the types. These definitions are essential for TypeScript to understand the types of libraries written in JavaScript.
+
+For example, when you use Node.js with TypeScript, you'll need type definitions for Node.js APIs. This is where `@types/node` comes in—it's a package that contains type definitions for Node.js standard library. You can install it using:
+
+```bash
+npm add --save-dev @types/node
+```
+
+These type definitions allow TypeScript to understand Node.js APIs and provide proper type checking and autocompletion when you use functions like `fs.readFile` or `http.createServer`. For example:
+
+```ts
+import * as fs from 'fs';
+
+// TypeScript knows the correct types thanks to @types/node
+fs.readFile('example.txt', 'utf8', (err, data) => {
+  if (err) throw err;
+  console.log(data);
+});
+```
```

Comment:
I think a more useful example would be an error-case:

````suggestion
```js
import * as fs from 'fs';

fs.readFile('example.txt', 'foo', (err, data) => {
//                          ^^^ Argument of type '"foo"' is not assignable to parameter of type …
  if (err) throw err;
  console.log(data);
});
```
````

Even though the above code is JavaScript, not TypeScript, the type definitions from `@types/node` provide the necessary information to determine that `'foo'` is not a valid value for the `encoding` parameter (a union type of specific string values like `'utf8'`).